src/api/contexts/auth.context.tsx

```diff
@@ -15,18 +17,25 @@ export function useAuthContext(): AuthInterface {
 
 export function AuthContextProvider(props: PropsWithChildren): JSX.Element {
   const [token, setToken] = useState<string>();
+  const [jwt, setJwt] = useState<Jwt>();
   const { authenticationToken } = useStore();
 
   const tokenWithFallback = token ?? authenticationToken.get();
-  const isLoggedIn = tokenWithFallback != undefined;
+  const isExpired = jwt?.exp != null && Date.now() > new Date(jwt?.exp * 1000).getTime();
+  const isLoggedIn = tokenWithFallback != undefined && !isExpired;
```

Comment:
The `isExpired` will not use tokenWithFallback 🤔  maybe do on-the-fly JWT decoding and only store the token?

Reply:
useEffect does use updateToken and every other change to token does use the same function, therefore JWT is always up-to-date. tokenWithFallback is only needed because of setState delaying the update in a render cycle.

Reply:
Hm, interesting. In which case happened this?

Reply:
I think I had the issue on initial load, that is not reflecting the correct state, that is why I have introduced tokenWithFallback. Maybe the issue is using token on two different layers
1) storage
2) state

Might be better to just remove the state and just read from storage all the time 🤷‍♂️